src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -245,21 +245,36 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
         return configs
     }
 
+    /**
+     * Necessary to get the debug configuration over to the resolveDebugConfigurationWithSubstitutedVariables function below
+     *
+     * @param folder Workspace folder
+     * @param config User-provided config (from launch.json)
+     * @param token  Cancellation token
+     */
+    public resolveDebugConfiguration(
+        folder: vscode.WorkspaceFolder | undefined,
+        config: AwsSamDebuggerConfiguration,
+        token?: vscode.CancellationToken
+    ): vscode.ProviderResult<AwsSamDebuggerConfiguration> {
+        return config
+    }
+
     /**
      * Generates a full run-config from a user-provided config, then
      * runs/debugs it (essentially `sam build` + `sam local invoke`).
      *
      * If `launch.json` is missing, attempts to generate a config dynamically.
      *
-     * @param folder  Workspace folder
+     * @param folder Workspace folder
      * @param config User-provided config (from launch.json)
      * @param token  Cancellation token
      */
-    public async resolveDebugConfiguration(
+    public async resolveDebugConfigurationWithSubstitutedVariables(
         folder: vscode.WorkspaceFolder | undefined,
         config: AwsSamDebuggerConfiguration,
         token?: vscode.CancellationToken
-    ): Promise<SamLaunchRequestArgs | undefined> {
+    ): Promise<undefined> {
```

Comment:
why change the return type?

Reply:
Two reasons:
1. This only returns `undefined`. I'm unsure if it ever returned that type.
2. The proper return type was a [debug configuration](https://code.visualstudio.com/api/references/vscode-api#DebugConfigurationProvider). This at the very least leaves out ambiguity that something would be doable with a `SamLaunchRequestArgs` type.

Reply:
k, but it will need to be changed back when the TODO is addressed https://github.com/aws/aws-toolkit-vscode/pull/1522#discussion_r578867964

Reply:
? That's not the proper implementation though. The proper thing is to create a separate debug adapter, which this will pass an `AwsSamDebuggerConfiguration` (or some other `vscode.DebugConfiguration`) to. The debug adapter should handle converting the debug configuration to SamLaunchRequestArgs.

Honestly, that means a correct implementation of this function will be to just pass the config through.

Reply:
I have a branch that implements the "correct" way. 

The type actually will be `SamLaunchRequestArgs`, though it doesn't matter really because the debugger only wants the fields in `AwsSamDebuggerConfiguration`.

> The debug adapter should handle converting the debug configuration to SamLaunchRequestArgs.

Nah. I tried that way originally, but it defeats the purpose, because then we need a custom DebugAdapter.